bond/bond.go

```diff
@@ -189,46 +189,46 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 	if netNs, err = ns.GetNS(nspath); err != nil {
 		return fmt.Errorf("failed to open netns %q: %v", nspath, err)
 	}
+	defer netNs.Close()
 
 	if curnetNs, err = ns.GetCurrentNS(); err != nil {
 		return fmt.Errorf("failed to get init netns: %v", err)
 	}
 
 	if releaseLinks {
-		if err := netNs.Set(); err != nil {
-			return fmt.Errorf("failed to enter netns %q: %v", netNs, err)
-		}
+		err = move(linkNames, netNs, curnetNs)
+	} else {
+		err = move(linkNames, curnetNs, netNs)
 	}
+	return err
+}
 
-	if len(linkNames) >= 2 { // currently supporting two or more links to one bond
-		for _, linkName := range linkNames {
-			// get interface link in the network namespace
-			link, err := netlink.LinkByName(linkName)
-			if err != nil {
-				return fmt.Errorf("failed to lookup link interface %q: %v", linkName, err)
-			}
-
-			// set link interface down
-			if err = netlink.LinkSetDown(link); err != nil {
-				return fmt.Errorf("failed to down link interface %q: %v", linkName, err)
-			}
+func move(links []string, from ns.NetNS, to ns.NetNS) error {
+	err := from.Do(func(ns.NetNS) error {
+		if len(links) >= 2 { // currently supporting two or more links to one bond
+			for _, linkName := range links {
+				// get interface link in the network namespace
+				link, err := netlink.LinkByName(linkName)
+				if err != nil {
+					return fmt.Errorf("failed to lookup link interface %q: %v", linkName, err)
+				}
 
-			if releaseLinks { // move link inteface to network netns
-				if err = netlink.LinkSetNsFd(link, int(curnetNs.Fd())); err != nil {
-					return fmt.Errorf("failed to move link interface to host netns %q: %v", linkName, err)
+				// set link interface down
+				if err = netlink.LinkSetDown(link); err != nil {
+					return fmt.Errorf("failed to down link interface %q: %v", linkName, err)
 				}
-			} else {
-				if err = netlink.LinkSetNsFd(link, int(netNs.Fd())); err != nil {
-					return fmt.Errorf("failed to move link interface to container netns %q: %v", linkName, err)
+
+				// move link inteface to network netns
+				if err = netlink.LinkSetNsFd(link, int(to.Fd())); err != nil {
+					return fmt.Errorf("failed to move link interface to netns %q: %v", linkName, err)
```

Comment:
let's use the releaseLinks variable here just for the logging message in case of error to differentiate between host namespace and container namespace.

Reply:
added another argument to the function containing the ns role (container/ host) and used it in the log message. the string value is passed according to the releaseLinks variable